File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.js
function solution(keymap, targets) {
    let result = [];
    let count;
    
    for (let i = 0; i < targets.length; i++) {
        count = 0;
        for (let j = 0; j < targets[i].length; j++) {
            let keyIndex = keymap.map((key) => {
                if (key.indexOf(targets[i][j]) === -1) {
                    return 99999;
                } else {
                    return key.indexOf(targets[i][j]);
                }
            })
            // console.log(targets[i][j], keyIndex);
            count += Math.min(...keyIndex) + 1;
        }
        if (count > 99999) {
            result.push(-1);
        } else {
            result.push(count);
        }
    }
    return result;
}

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.py
def solution(keymap, targets):
    result = []
    
    for target in targets:
        count = 0
        for j in target:
            min = 99999
            for key in keymap:
                if (key.find(j) != -1 and key.find(j) < min):
                    min = key.find(j)
            count += min + 1
        if count > 9999:
            result.append(-1)
        else:
            result.append(count)
    return result